crates/interpreter/src/instructions/opcode.rs

```diff
@@ -399,584 +605,120 @@ opcodes! {
-        0xB6 => OpInfo::none(),
-        0xB7 => OpInfo::none(),
-        0xB8 => OpInfo::none(),
-        0xB9 => OpInfo::none(),
-        0xBA => OpInfo::none(),
-        0xBB => OpInfo::none(),
-        0xBC => OpInfo::none(),
-        0xBD => OpInfo::none(),
-        0xBE => OpInfo::none(),
-        0xBF => OpInfo::none(),
-        0xC0 => OpInfo::none(),
-        0xC1 => OpInfo::none(),
-        0xC2 => OpInfo::none(),
-        0xC3 => OpInfo::none(),
-        0xC4 => OpInfo::none(),
-        0xC5 => OpInfo::none(),
-        0xC6 => OpInfo::none(),
-        0xC7 => OpInfo::none(),
-        0xC8 => OpInfo::none(),
-        0xC9 => OpInfo::none(),
-        0xCA => OpInfo::none(),
-        0xCB => OpInfo::none(),
-        0xCC => OpInfo::none(),
-        0xCD => OpInfo::none(),
-        0xCE => OpInfo::none(),
-        0xCF => OpInfo::none(),
-        0xD0 => OpInfo::none(),
-        0xD1 => OpInfo::none(),
-        0xD2 => OpInfo::none(),
-        0xD3 => OpInfo::none(),
-        0xD4 => OpInfo::none(),
-        0xD5 => OpInfo::none(),
-        0xD6 => OpInfo::none(),
-        0xD7 => OpInfo::none(),
-        0xD8 => OpInfo::none(),
-        0xD9 => OpInfo::none(),
-        0xDA => OpInfo::none(),
-        0xDB => OpInfo::none(),
-        0xDC => OpInfo::none(),
-        0xDD => OpInfo::none(),
-        0xDE => OpInfo::none(),
-        0xDF => OpInfo::none(),
-        0xE0 => OpInfo::none(),
-        0xE1 => OpInfo::none(),
-        0xE2 => OpInfo::none(),
-        0xE3 => OpInfo::none(),
-        0xE4 => OpInfo::none(),
-        0xE5 => OpInfo::none(),
-        0xE6 => OpInfo::none(),
-        0xE7 => OpInfo::none(),
-        0xE8 => OpInfo::none(),
-        0xE9 => OpInfo::none(),
-        0xEA => OpInfo::none(),
-        0xEB => OpInfo::none(),
-        0xEC => OpInfo::none(),
-        0xED => OpInfo::none(),
-        0xEE => OpInfo::none(),
-        0xEF => OpInfo::none(),
-        CREATE => OpInfo::gas_block_end(0),
-        CALL => OpInfo::gas_block_end(0),
-        CALLCODE => OpInfo::gas_block_end(0),
-        RETURN => OpInfo::gas_block_end(0),
-        DELEGATECALL => OpInfo::gas_block_end(0),
-        CREATE2 => OpInfo::gas_block_end(0),
-        0xF6 => OpInfo::none(),
-        0xF7 => OpInfo::none(),
-        0xF8 => OpInfo::none(),
-        0xF9 => OpInfo::none(),
-        STATICCALL => OpInfo::gas_block_end(0),
-        0xFB => OpInfo::none(),
-        0xFC => OpInfo::none(),
-        REVERT => OpInfo::gas_block_end(0),
-        INVALID => OpInfo::gas_block_end(0),
-        SELFDESTRUCT => OpInfo::gas_block_end(0),
-    }
-}
-
-const fn make_gas_table(spec: SpecId) -> [OpInfo; 256] {
-    let mut table = [OpInfo::none(); 256];
-    let mut i = 0;
-    while i < 256 {
-        table[i] = opcode_gas_info(i as u8, spec);
-        i += 1;
-    }
-    table
-}
-
-/// Returns a lookup table of opcode gas info for the given [`SpecId`].
-#[inline]
-pub const fn spec_opcode_gas(spec_id: SpecId) -> &'static [OpInfo; 256] {
```

Comment:
How does this map onto the new opcode info? We need a way to get the static gas of opcodes for a specific SpecId

Reply:
I checked foundry, and it is only used to check if it is push opcode or not